res/controllers/midi-controls-0.0.js

```diff
@@ -0,0 +1,668 @@
+        this.outCo = 'hotcue_' + this.number + '_enabled';
+        Button.call(this, options);
+    };
+    HotcueButton.prototype = new Button({
+        unshift: function () {
+            this.inCo = 'hotcue_' + this.number + '_activate';
+        },
+        shift: function () {
+            this.inCo = 'hotcue_' + this.number + '_clear';
+        },
+        onlyOnPress: false,
+    });
+
+    var SamplerButton = function (options) {
+        if (options.number === undefined) {
+            print('WARNING: No sampler number specified for new SamplerButton.');
+        }
+        this.number = options.number;
+        this.group = '[Sampler' + this.number + ']';
+        Button.call(this, options);
+    };
+    SamplerButton.prototype = new Button({
+        unshift: function () {
+            this.input = function (channel, control, value, status, group) {
+                if (value > 0) {
+                    if (engine.getValue(this.group, 'track_loaded') === 0) {
+                        engine.setValue(this.group, 'LoadSelectedTrack', 1);
+                    } else {
+                        engine.setValue(this.group, 'cue_gotoandplay', 1);
+                    }
+                }
+            };
+        },
+        shift: function() {
+            this.input = function (channel, control, value, status, group) {
+                if (value > 0) {
+                    if (engine.getValue(this.group, 'play') === 1) {
+                        engine.setValue(this.group, 'play', 0);
+                    } else {
+                        engine.setValue(this.group, 'eject', 1);
+                    }
+                }
+            };
+        },
+        output: function (value, group, control) {
+            if (engine.getValue(this.group, 'track_loaded') === 1) {
+                if (this.playing === undefined) {
+                    this.send(this.on);
+                } else {
+                    if (engine.getValue(this.group, 'play') === 1) {
+                        this.send(this.on);
+                    } else {
+                        this.send(this.playing);
+                    }
+                }
+            } else {
+                this.send(this.off);
+            }
+        },
+        connect: function() {
+            this.connections[0] = engine.connectControl(this.group, 'track_loaded', this.output);
+            if (this.playing !== undefined) {
+                this.connections[1] = engine.connectControl(this.group, 'play', this.output);
+            }
+        },
+        outCo: null, // hack to get Control constructor to call connect()
+    });
+
+    var Pot = function (options) {
+        Control.call(this, options);
+
+        this.firstValueReceived = false;
+    };
+    Pot.prototype = new Control({
+        inValueScale: function (value) { return value / this.max; },
+        input: function (channel, control, value, status, group) {
+            this.inSetParameter(this.inValueScale(value));
```

Comment:
This cannot properly handle pots that use 14 bit MIDI signals. I don't have any controllers that use 14 bit MIDI signals, so if you do, contributions are welcome.